gcp/disk.go

```diff
@@ -51,6 +52,9 @@ func (d *Disk) LastUsedAt() time.Time {
 // SizeGB returns the size of the GCP compute disk in GB.
 func (d *Disk) SizeGB() int { return int(d.Disk.SizeGb) }
 
+// SizeBytes returns the size of the GCP compute disk in bytes.
+func (d *Disk) SizeBytes() int { return int(d.Disk.SizeGb) * int(math.Pow(1000, 3)) }
```

Comment:
Nit: Providers charge per GiB, which is `1024 ^3`

Reply:
In addition to being simpler to read, using `math.Pow` is significantly less performant:

```
inkel@numenor:~/go/src/github.com/inkel/benchmarks$ go test -bench=BenchmarkPow -benchmem
goos: darwin
goarch: arm64
pkg: github.com/inkel/benchmarks
BenchmarkPow/1000*1000*1000-10          1000000000               0.3217 ns/op          0 B/op          0 allocs/op
BenchmarkPow/math.Pow-10                125777395                9.544 ns/op           0 B/op          0 allocs/op
PASS
ok      github.com/inkel/benchmarks     2.779s
```

Reply:
> Nit: Providers charge per GiB, which is 1024 ^3

Where did you find that? 👀  I tried to check and all the pricing I found was in GB. Even the AWS one, even though they provide the size of the volumes on GiB.

Reply:
They don't make it easy and I only know this because I've spent way too much time reading the pricing pages for each CSP. 

---

Google: https://cloud.google.com/compute/disks-image-pricing#disk

> For Compute Engine, disk size, machine type memory, and network usage are calculated in JEDEC binary gigabytes (GB), or IEC gibibytes (GiB), where 1 GiB is 230 bytes. Similarly, 1 TiB is 240 bytes, or 1024 JEDEC GBs.

---

AWS: https://aws.amazon.com/s3/pricing/

---

Azure: https://azure.microsoft.com/en-us/pricing/details/storage/blobs/#pricing

> 

Azure Data Lake Storage usage is calculated in binary Gigabytes (GB), where 1 GB = 230 bytes. This unit of measurement is also known as Gibibyte (GiB), defined by the International Electrotechnical Commission (IEC). Similarly, 1 TB is 240 bytes, i.e. 1,024 GBs.




> Amazon S3 storage usage is calculated in binary gigabytes (GB), where 1 GB is 230 bytes. This unit of measurement is also known as a gibibyte (GiB), defined by the International Electrotechnical Commission (IEC). Similarly, 1 TB is 240 bytes, i.e. 1024 GBs.

Reply:
Am I the only one finding AWS and Azure text TOO similar, as in one copied the other and only changed their product name?

<img src="https://media1.giphy.com/media/l0IykOsxLECVejOzm/giphy.gif"/>